### PR TITLE
Javascript/javascript2/week1

### PR DESCRIPTION
**Pull request by Enqira**
Finished HW js2 week1

### CODE REVIEW

File: javascript/javascript2/week1/exercise-1/cardio.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+function findShortestWord(danishWords) {
+  let shortestWord = "";
+  for (let i = 0; i < danishWords.length; i++) {
+    if (shortestWord === "") {
+      shortestWord = danishWords[i];
+      console.log(shortestWord.length);
```
Comment (Rassibassi) [inline javascript/javascript2/week1/exercise-1/cardio.js:7]: this `if` clause is not really necessary, why not setting `shortestWord = danishWords[0]` in line 3 ? Can you improve your code with that change and see how it simplifies?

File: javascript/javascript2/week1/exercise-1/danishLettersCounter.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  for (let i = 0; i < danishString.length; i++) {
+    if (danishString[i] === "å") {
+      if (result.hasOwnProperty("å")) {
+        result.å++;
+        result.total++;
+      } else {
+        result.å = 1;
+        result.total++;
+      }
+    }
```
Comment (Rassibassi) [inline javascript/javascript2/week1/exercise-1/danishLettersCounter.js:18]: If letter is `'å'` then your code will always also check if it is `'æ'` and `'ø'`. Can you somehow avoid that? Hint: `else if`

File: javascript/javascript2/week1/exercise-1/danishLettersCounter.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const danishString2 = "Blå grød med røde bær";
+
+function countDanishLetters(danishString) {
+  let result = { total: 0 };
+  let å = 0;
+  let æ = 0;
+  let ø = 0;
+  for (let i = 0; i < danishString.length; i++) {
+    if (danishString[i] === "å") {
+      if (result.hasOwnProperty("å")) {
```
Comment (Rassibassi) [inline javascript/javascript2/week1/exercise-1/danishLettersCounter.js:11]: `result.hasOwnProperty("å")` is simpler this way: `"å" in result`

File: javascript/javascript2/week1/exercise-1/danishLettersCounter.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        result.å++;
+        result.total++;
+      } else {
+        result.å = 1;
+        result.total++;
```
Comment (Rassibassi) [inline javascript/javascript2/week1/exercise-1/danishLettersCounter.js:16]: `result.total++` can go outside the `if else` here

File: javascript/javascript2/week1/hyfBay/main.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+function renderProducts(products) {
+  for (let i = 0; i < products.length; i++) {
+    //   select and create tags
+    const ulTag = document.querySelector("ul");
+    const liTag = document.createElement("li");
+    const nameTag = document.createElement("h3");
+    const pricetag = document.createElement("span");
+    const ratingTag = document.createElement("span");
+    const br = document.createElement("br");
```
Comment (Rassibassi) [inline javascript/javascript2/week1/hyfBay/main.js:14]: make the `span` to `div` and no `br` is needed

File: javascript/javascript2/week1/name-generator/index.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    <title>Document</title>
+</head>
+
+<body>
+    <div>
+        <input id="user-input" placeholder="Write your name here...">
+        <button id="button" onclick="btnClicked()" style="display: block;">Accept</button>
+        <p id="displayed-text"></p>
+        <br>
+        <button id="sec-button" onclick="secBtnClicked()" style="display: none;">One more spirit please</button>
```
Comment (Rassibassi) [inline javascript/javascript2/week1/name-generator/index.html:17]: Please use `addEventListener` in the javascript to add an event to a dom element. intead of in the html

**Review by Rassibassi — COMMENTED** [review]
Good work, let me know if you have any questions regarding my comments

File: javascript/javascript2/week1/exercise-1/cardio.js (hunk 1 above)
Comment (Enqira) [reply javascript/javascript2/week1/exercise-1/cardio.js:7]: Got it, Thnak you

File: javascript/javascript2/week1/exercise-1/danishLettersCounter.js (hunk 2 above)
Comment (Enqira) [reply javascript/javascript2/week1/exercise-1/danishLettersCounter.js:18]: You are right, I fixed that :)

File: javascript/javascript2/week1/exercise-1/danishLettersCounter.js (hunk 3 above)
Comment (Enqira) [reply javascript/javascript2/week1/exercise-1/danishLettersCounter.js:11]: Much simpler, Thank you

**Pull request merged**